resources/scponly_user.rb

```diff
@@ -0,0 +1,126 @@
+# true To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponly_user
+
+default_action :create
+
+property :write_dir, String, default: 'incoming'
+property :public_key, String
+property :private_key, String
+property :chroot, [true, false], default: true
+property :altroot, String
+property :binaries,
+          Array,
+          default:
+          %w(
+            /bin/chgrp
+            /bin/chmod
+            /bin/chown
+            /bin/ln
+            /bin/ls
+            /bin/mkdir
+            /bin/mv
+            /bin/rm
+            /bin/rmdir
+            /usr/bin/scp
+            /usr/sbin/scponlyc
+          )
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  if new_resource.chroot
+
+    altroot = new_resource.altroot.nil? ? '/home/chroot' : new_resource.altroot
+
+    directory "#{altroot}/home/#{new_resource.name}" do
+      recursive true
+    end
+
+    user new_resource.name do
+      gid 'scponly'
+      manage_home true
+      home "#{altroot}/home/#{new_resource.name}"
+      shell '/usr/sbin/scponlyc'
+    end
+
+    directory "#{altroot}/home/#{new_resource.name}/#{new_resource.write_dir}" do
+      owner 'root'
+      group 'scponly'
+      mode '0770'
+      recursive true
+    end
+
+    directory '/usr/share/libexec/'
+    cookbook_file '/usr/share/libexec/scponly-chroot.sh' do
```

Comment:
I told you the wrong directory unfortunately. Please use this instead and update the resource below.
```suggestion
    cookbook_file '/usr/libexec/scponly-chroot.sh' do
```